doc/troubleshooting.md

```diff
@@ -0,0 +1,362 @@
+`autoApplyRemediations=true`. Typically, the administrator would now review
+the remediations and apply them as needed.
+
+If the remediations are set to be applied automatically, the ComplianceSuite
+controller takes over in the Done phase, pauses the `MachineConfigPool` to
+which the scan maps to and applies all the remediations in one go.
+
+Either way, if a remediation is applied, the `ComplianceRemediation`
+controller takes over.
+
+## ComplianceRemediation lifecycle and debugging
+At this point, our example scan has reported some findings. We can enable
+one of the remediations by toggling its `apply` attribute to `true`:
+```
+oc patch complianceremediations/rhcos4-e8-worker-audit-rules-dac-modification-chmod --patch '{"spec":{"apply":true}}' --type=merge
+```
+
+The complianceremediation controller (`logger=remediationctrl`) reconciles
+the modified object. The result of the reconciliation is change of status of
+the remediation object that is reconciled, but also a change of the rendered
+per-suite `MachineConfig` object that contains all the applied remediations.
+
+The `MachineConfig` object always begins with `75-` and is named after the
+scan and the suite, in our case it would be:
+```
+oc get mc | grep 75-
+75-rhcos4-e8-worker-my-companys-compliance-requirements                                                2.2.0             2m46s
+```
+
+The remediations the MC currently consists of are listed in the MC's
+annotations:
+```
+oc describe mc/75-rhcos4-e8-worker-my-companys-compliance-requirements
+Name:         75-rhcos4-e8-worker-my-companys-compliance-requirements
+Labels:       machineconfiguration.openshift.io/role=worker
+Annotations:  remediation/rhcos4-e8-worker-audit-rules-dac-modification-chmod: 
+```
+
+The complianceremediation controller's algorithm works roughly like this:
+
+   * All currently applied remediations are read
+   * If the reconciled remediation is supposed to be applied, it is added
+     to the set
+   * A `MachineConfig` object is rendered from the set and annotated with names
+     of remediations in the set. If the set is empty (the last remediation
+     was unapplied), the rendered MachineConfig object is removed.
+   * If and only if the rendered `MachineConfig` is different from the one
+     already applied in the cluster, the applied MC is updated (or created,
+     or deleted).
+   * Creating or modifying a MachineConfig object triggers a reboot of nodes
+     that match the `machineconfiguration.openshift.io/role` label - see the
+     MachineConfig Operator documentation for more details.
+
+The remediation loop ends once the rendered MachineConfig is updated, if needed,
+and the reconciled remediation object status is updated.
+
+In our case, applying the remediation would trigger a reboot. After that we can
+annotate the scan to re-run it:
+```
+oc annotate compliancescans/rhcos4-e8-worker compliance.openshift.io/rescan
```

Comment:
The command would be :

> $ oc annotate compliancescans/worker-scan compliance.openshift.io/rescan=
> compliancescan.compliance.openshift.io/worker-scan annotated

  to annotate the scan at line 290 & 354 otherwise, it reports the following error :

> $ oc annotate compliancescans/worker-scan compliance.openshift.io/rescan
> error: at least one annotation update is required

Reply:
good catch, fixed